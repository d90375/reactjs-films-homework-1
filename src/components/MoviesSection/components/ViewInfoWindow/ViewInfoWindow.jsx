import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../../Button';

import styles from './ViewInfoWindow.scss';

const ViewInfoWindow = ({
  film: {
    title, genres, rating, overview,
  },
}) => (
  <div className={styles.container}>
    <button className={styles.close} type="button">&#215;</button>
    <div className={styles.info}>
      <h3 className={styles.title}>
        {title.length < 18 ? title : `${title.substring(0, 15)}...`}
      </h3>
      <p className={styles.rating}>{rating}</p>
      <p className={styles.genres}>{genres}</p>
    </div>
    <p className={styles.overview}>{overview}</p>
    <Button color="primary">Watch Now</Button>
  </div>
);

ViewInfoWindow.propTypes = {
  film: PropTypes.shape({
    title: PropTypes.string,
    genres: PropTypes.string,
    rating: PropTypes.number,
    overview: PropTypes.string,
  }).isRequired,
};

export default ViewInfoWindow;
