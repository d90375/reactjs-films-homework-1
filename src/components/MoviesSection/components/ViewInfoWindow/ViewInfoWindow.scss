@import '../../../../mixins.scss';

.container {
  position: absolute;
  top: 0;
  width: 250px;
  height: 320px;
  padding: 10px 20px;
  border-radius: 6px;
  background-color: rgba($color: #000, $alpha: 0.7);
  box-sizing: border-box;
  display: none;
}

.close {
  font-size: 22px;
  background-color: transparent;
  color: white;
  border: none;
  border-radius: 3px;

  @include outline-none;

  &:hover {
    background-color: rgba($color: #000, $alpha: 0.8);
  }
}

.info {
  box-sizing: border-box;
  width: 100%;
  height: 85px;

  .title {
    display: inline-block;
    width: 170px;
    margin-bottom: 10px;
    font-size: 16px;
    line-height: 22px;
    font-weight: bold;
    text-transform: uppercase;
    color: white;
  }

  .rating {
    display: inline-block;
    width: 30px;
    margin-bottom: 10px;
    font-size: 12px;
    line-height: 20px;
    font-weight: bold;
    color: #47c3f0;
    text-align: center;
    border-radius: 6px;
    border: 2px solid #49c1f0;
  }

  .genres {
    margin: 0;
    font-size: 12px;
    line-height: 16px;
    letter-spacing: 1px;
    color: #47c3f0;
    font-weight: bold;
  }
}

.overview {
  height: 110px;
  font-size: 14px;
  line-height: 22px;
  color: white;
  overflow: hidden;
}
