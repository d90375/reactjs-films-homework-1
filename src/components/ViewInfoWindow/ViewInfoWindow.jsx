import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import styles from './ViewInfoWindow.scss';

const ViewInfoWindow = ({
  film: {
    id, title, genres, rating, overview, poster,
  },
  switchViewInfo,
  fetchTrailer,
}) => (
  <div className={styles.container}>
    <img className={styles.poster} src={poster} alt="poster" />
    <div className={styles.wrapper}>
      <button className={styles.close} type="button" onClick={switchViewInfo}>&#215;</button>
      <div className={styles.info}>
        <h3 className={styles.title}>
          {title.length < 18 ? title : `${title.substring(0, 15)}...`}
        </h3>
        <p className={styles.rating}>{rating}</p>
        <p className={styles.genres}>{genres}</p>
      </div>
      <p className={styles.overview}>{overview}</p>
      <Button color="primary" onClick={() => fetchTrailer(id)}>Watch Now</Button>
    </div>
  </div>
);

ViewInfoWindow.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    genres: PropTypes.string,
    rating: PropTypes.number,
    overview: PropTypes.string,
    poster: PropTypes.string,
  }).isRequired,
  switchViewInfo: PropTypes.func.isRequired,
  fetchTrailer: PropTypes.func.isRequired,
};

export default ViewInfoWindow;
