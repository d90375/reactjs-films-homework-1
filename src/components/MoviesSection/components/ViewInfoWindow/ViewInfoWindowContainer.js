import { connect } from 'react-redux';
import { fetchTrailer } from '../../../../modules/trailer/trailerAction';
import ViewInfoWindow from './ViewInfoWindow';

const mapDispatchToProps = {
  fetchTrailer,
};

export default connect(
  null,
  mapDispatchToProps,
)(ViewInfoWindow);
