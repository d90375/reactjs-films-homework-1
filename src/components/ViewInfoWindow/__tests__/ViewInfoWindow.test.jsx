import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';
import ViewInfoWindow from '../ViewInfoWindow';

describe('ViewInfoWindow tests', () => {
  const film = {
    id: 123,
    title: 'title',
    genres: 'action, crime',
    rating: 8,
    overview: 'overview',
    poster: 'url',
  };

  describe('ViewInfoWindow render', () => {
    it('ViewInfoWindow renders correctly with movie title length less then 18 characters', () => {
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<ViewInfoWindow
        switchViewInfo={mockCallBack}
        fetchTrailer={mockCallBack}
        film={film}
      />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });

    it('ViewInfoWindow renders correctly with movie title length more then 18 characters', () => {
      const mockCallBack = jest.fn();
      film.title = 'title!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!';
      const renderer = new ShallowRenderer();
      renderer.render(<ViewInfoWindow
        switchViewInfo={mockCallBack}
        fetchTrailer={mockCallBack}
        film={film}
      />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });

  describe('onClick function', () => {
    it('onClick callback function called', () => {
      const mockCallBack = jest.fn();

      const tree = create(
        <ViewInfoWindow
          switchViewInfo={mockCallBack}
          fetchTrailer={mockCallBack}
          film={film}
        >
          Trending
        </ViewInfoWindow>,
      );

      const button = tree.root.findByProps({ color: 'primary' });
      button.props.onClick();

      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack).toHaveBeenCalledWith(123);
    });
  });
});
