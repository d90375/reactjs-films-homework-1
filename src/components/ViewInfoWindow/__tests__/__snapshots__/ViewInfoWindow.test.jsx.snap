// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewInfoWindow tests ViewInfoWindow render ViewInfoWindow renders correctly 1`] = `
<div
  className="container"
>
  <img
    alt="poster"
    className="poster"
    src="url"
  />
  <div
    className="wrapper"
  >
    <button
      className="close"
      onClick={[MockFunction]}
      type="button"
    >
      ×
    </button>
    <div
      className="info"
    >
      <h3
        className="title"
      >
        title
      </h3>
      <p
        className="rating"
      >
        8
      </p>
      <p
        className="genres"
      >
        action, crime
      </p>
    </div>
    <p
      className="overview"
    >
      overview
    </p>
    <Button
      color="primary"
      onClick={[Function]}
    >
      Watch Now
    </Button>
  </div>
</div>
`;

exports[`ViewInfoWindow tests ViewInfoWindow render ViewInfoWindow renders correctly 2`] = `
<div
  className="container"
>
  <img
    alt="poster"
    className="poster"
    src="url"
  />
  <div
    className="wrapper"
  >
    <button
      className="close"
      onClick={[MockFunction]}
      type="button"
    >
      ×
    </button>
    <div
      className="info"
    >
      <h3
        className="title"
      >
        title!!!!!!!!!!...
      </h3>
      <p
        className="rating"
      >
        8
      </p>
      <p
        className="genres"
      >
        action, crime
      </p>
    </div>
    <p
      className="overview"
    >
      overview
    </p>
    <Button
      color="primary"
      onClick={[Function]}
    >
      Watch Now
    </Button>
  </div>
</div>
`;
